cosmos/executor.go

```diff
@@ -0,0 +1,457 @@
+package cosmos
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/dan13ram/wpokt-validator/app"
+	"github.com/dan13ram/wpokt-validator/common"
+	cosmos "github.com/dan13ram/wpokt-validator/cosmos/client"
+	"github.com/dan13ram/wpokt-validator/models"
+
+	"github.com/dan13ram/wpokt-validator/cosmos/util"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	multisigtypes "github.com/cosmos/cosmos-sdk/crypto/types/multisig"
+
+	signingtypes "github.com/cosmos/cosmos-sdk/types/tx/signing"
+)
+
+const (
+	BurnExecutorName = "BURN EXECUTOR"
+)
+
+type BurnExecutorRunner struct {
+	signer       *app.PocketSigner
+	client       cosmos.CosmosClient
+	wpoktAddress string
+	vaultAddress string
+}
+
+func (x *BurnExecutorRunner) Run() {
+	x.SyncTxs()
+}
+
+func (x *BurnExecutorRunner) Status() models.RunnerStatus {
+	return models.RunnerStatus{}
+}
+
+var utilValidateSignature = util.ValidateSignature
+var multisigtypesAddSignatureV2 = multisigtypes.AddSignatureV2
+
+func (x *BurnExecutorRunner) ValidateSignaturesAndAddMultiSignatureToTxConfig(
+	originTxHash string,
+	sequence uint64,
+	txCfg client.TxConfig,
+	txBuilder client.TxBuilder,
+) bool {
+	logger := log.
+		WithField("tx_hash", originTxHash).
+		WithField("section", "validate-signatures")
+
+	tx := txBuilder.GetTx()
+
+	sigV2s, err := tx.GetSignaturesV2()
+	if err != nil {
+		logger.WithError(err).Error("Error getting signatures")
+		return false
+	}
+
+	if len(sigV2s) == 0 || len(sigV2s) < int(x.signer.Multisig.GetThreshold()) {
+		logger.Errorf("Not enough signatures")
+		return false
+	}
+
+	account, err := x.client.GetAccount(x.signer.MultisigAddress)
+
+	if err != nil {
+		logger.WithError(err).Error("Error getting account")
+		return false
+	}
+
+	multisigSig := multisigtypes.NewMultisig(len(x.signer.Multisig.GetPubKeys()))
```

Comment:
I believe we should check if `sig.pubKey` exists inside `x.signer.Multisig.GetPubKeys()`.